### PR TITLE
Added tags update route. Added sort to user service

### PR DESCRIPTION
**Pull request by TheHollidayInn**
#7312
### Changes

Update user service to use sort tags
Update tag service to send update tags route
Added update tags route and tests

---

UUID:

### CODE REVIEW

**Comment by paglias** [comment]
Hmm not sure about this one.. I think a moveTag route would be better like we have for tasks

**Comment by TheHollidayInn** [comment]
Sounds good. I figured it could go either way. Ill update tonight.

**Comment by paglias** [comment]
Sorry if i'm being picky on this but for tasks we only ask for the task id (tag id in this case) and `to` parameter and I'd like it to work the same way. Then you get the original position by searching for the position of the id in the array

**Comment by TheHollidayInn** [comment]
No worries! I was just staying consistent with the user ops, but that makes more sense to be consistent with the tasks.

File: website/server/controllers/api-v3/tags.js [hunk 1]
```diff
@@ -114,6 +114,41 @@ api.updateTag = {
...
+    req.checkBody('to', res.t('toRequired')).notEmpty();
+    req.checkBody('tagId', res.t('tagIdRequired')).notEmpty();
+
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let tagIndex = _.findIndex(user.tags, function findTag (tag) {
+      return tag.id === req.body.tagId;
+    });
+    if (tagIndex === !1) throw new NotFound(res.t('tagNotFound'));
```
Comment (paglias) [inline website/server/controllers/api-v3/tags.js]: hmm === !1? I think you mean === -1

Comment (TheHollidayInn) [inline website/server/controllers/api-v3/tags.js]: Meh.. Late nights. Let me try again with and add a test.

**Comment by paglias** [comment]
The failing test is the usual one that always fails
